File: easy/compare_points.lua
for line in io.lines(arg[1]) do
  local x1, y1, x2, y2 = line:match("(-?%d+) (-?%d+) (-?%d+) (-?%d+)")
  x1, y1, x2, y2 = tonumber(x1), tonumber(y1), tonumber(x2), tonumber(y2)
  print((x1 == x2) and ((y1 == y2) and "here" or
                        (y1 < y2) and "N" or "S") or
        (y1 == y2) and ((x1 < x2) and "E" or "W") or
        ((x1 < x2) and ((y1 < y2) and "NE" or "SE") or
                       ((y1 < y2) and "NW" or "SW")))
end

File: easy/compare_points.clj
(require '[clojure.string :as str])
(doseq [item (for [st (str/split-lines (slurp (first *command-line-args*)))]
  (when true
    (def p (map #(Integer/parseInt (str %)) (str/split st #" ")))
    (cond
      (= (first p) (nth p 2))
        (if (= (second p) (nth p 3))
          "here"
          (if (< (second p) (nth p 3))
            "N"
            "S"))
      (= (second p) (nth p 3))
        (if (< (first p) (nth p 2))
          "E"
          "W")
      :else
        (if (< (first p) (nth p 2))
          (if (< (second p) (nth p 3))
            "NE"
            "SE")
          (if (< (second p) (nth p 3))
            "NW"
            "SW"))))
)] (println item))
